website/src/components/Messages/MessageTree.tsx

```diff
@@ -9,9 +9,10 @@ const messagePaddingTop = 16;
 const avatarSize = 32;
 const avartarMarginTop = 6;
 const maxDepth = 100; // this only used for debug UI in mobile
-const left = avatarSize / 2 - 1;
+const toPx = (val: number) => `${val}px`;
 
-export const MessageTree = ({ tree, messageId }: { tree: MessageWithChildren; messageId?: string }) => {
+// eslint-disable-next-line react/display-name
+export const MessageTree = memo(({ tree, messageId }: { tree: MessageWithChildren; messageId?: string }) => {
   const renderChildren = (children: MessageWithChildren[], depth = 1) => {
```

Comment:
Should this be its own component? otherwise maybe useMemo or similar? because this will be re-created and re-rendered every time.

Reply:
It only rerenders if tree and messageId changed because we already have memo